// containers/quotations/index.js
import React, {Component} from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';

import { fetchQuotations } from 'actions';

class Quotations extends Component {
    componentDidMount() {
        this.props.fetchQuotations()
    }

    renderQuotations(quotation, index) {
        return (
            <div key={index}>
                <h4>{quotation.name}</h4>
            </div>
        )
    }

    render() {
        const pairs = this.props.pair;
        const listItems = pairs.map((pair) =>
            <li key={pair}>{pair}</li>
        );
        console.log(this.props.quotations)
        return (
            <div>
                {listItems}
            </div>
        )
    }
} 

Quotations.propTypes = {
    fetchQuotations: PropTypes.func.isRequired,
    pair: PropTypes.array.isRequired,
    quotations: PropTypes.object
};

const mapStateToProps = state => ({
    pair: state.quotationsPage.pair,
    quotations: state.quotations.items
})

const mapDispatchToProps = {
    fetchQuotations
}

export default connect(mapStateToProps, mapDispatchToProps)(Quotations);
